File: src/leetcode/2026/05/Solution888.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class Solution888 {
    public int[] fairCandySwap(int[] aliceSizes, int[] bobSizes) {
        int sumAlice = 0;
        for (int size : aliceSizes) {
            sumAlice += size;
        }

        int sumBob = 0;
        for (int size : bobSizes) {
            sumBob += size;
        }

        int diff = (sumAlice - sumBob) / 2;

        Set<Integer> bobSet = new HashSet<>();
        for (int size : bobSizes) {
            bobSet.add(size);
        }

        for (int size : aliceSizes) {
            if (bobSet.contains(size - diff)) {
                return new int[] { size, size - diff };
            }
        }

        return new int[0]; // This line should never be reached
    }

    public static void main(String[] args) {
        Solution888 solution = new Solution888();

        int[] aliceSizes1 = { 1, 1 };
        int[] bobSizes1 = { 2, 2 };
        System.out.println(Arrays.toString(solution.fairCandySwap(aliceSizes1, bobSizes1))); // Output: [1, 2]

        int[] aliceSizes2 = { 1, 2 };
        int[] bobSizes2 = { 2, 3 };
        System.out.println(Arrays.toString(solution.fairCandySwap(aliceSizes2, bobSizes2))); // Output: [1, 2]

        int[] aliceSizes3 = { 2 };
        int[] bobSizes3 = { 1, 3 };
        System.out.println(Arrays.toString(solution.fairCandySwap(aliceSizes3, bobSizes3))); // Output: [2, 3]
    }
}

File: src/leetcode/2026/05/Solution888.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <numeric>
using namespace std;

class Solution888 {
public:
    vector<int> fairCandySwap(vector<int>& aliceSizes, vector<int>& bobSizes) {
        int sumAlice = accumulate(aliceSizes.begin(), aliceSizes.end(), 0);
        int sumBob = accumulate(bobSizes.begin(), bobSizes.end(), 0);
        int diff = (sumAlice - sumBob) / 2;

        unordered_set<int> bobSet(bobSizes.begin(), bobSizes.end());
        for (int x : aliceSizes) {
            if (bobSet.count(x - diff)) {
                return {x, x - diff};
            }
        }
        return {};
    }
};

int main() {
    Solution888 solution;

    vector<int> aliceSizes = {1, 2, 5};
    vector<int> bobSizes = {2, 4};
    vector<int> result = solution.fairCandySwap(aliceSizes, bobSizes);
    cout << result[0] << " " << result[1] << endl; // Output: "5 4"

    aliceSizes = {1, 1};
    bobSizes = {2, 2};
    result = solution.fairCandySwap(aliceSizes, bobSizes);
    cout << result[0] << " " << result[1] << endl; // Output: "1 2"
    
    return 0;
}